defaults/main.yml

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: MIT
+---
+# The provider for nbde_client role. Currently 'clevis' is supported.
+nbde_client_provider: clevis
+
+# nbde_client_bindings - the description of the devices, that may also contain
+# an auth configuration for each group of devices.
+#
+# nbde_client_bindings supports the following keys:
+#  * devices - specifies LUKS devices, with their path and possibly either a
+#              passphrase or key file.
+#  * auth    - the settings for the binding, like the servers it should bind
+#              to.
+#  * state   - specifies whether the binding with the configuration described
+#              should be added or removed. Setting state to present (the
+#              default) means bindings will be added; setting state to absent
+#              means bindings will be removed from the devices/slots.
+#
+#
+# devices supports the following keys:
+# * path    - specifies the path to access this device (REQUIRED).
+# * pass    - specifies a valid passphrase for this device.
+# * keyfile - specifies either the absolute or relative path of a valid key
+#             file for this device.
+#
+# Example:
+# - devices:
+#    - path: /dev/sda1
+#      pass: password
+#    - path: /dev/sda2
+#      keyfile: /vault/keyfile
+#
+#
+# auth supports the following keys:
+# * slot               - specifies the slot to use for the binding. Default
+#                        is 1.
+# * servers            - specifies a list of servers to bind to. To enable
+#                        high availability, simply specify more than one server
+#                        here.
+# * threshold          - specifies the threshold for the Shamir Secret Sharing
+#                        (SSS) scheme that is put in place when using more
+#                        than one server. When using multiple servers,
+#                        threshold indicates how many of those servers should
+#                        succeed in order to complete the decryption.
+# * overwrite          - specifies whether we should override any existing
+#                        binding in the specified slot.
+# * discard_passphrase - specifies whether we should discard the passphrase
+#                        provided -- via either pass or keyfile, in devices
+#                        -- after completing the binding operation.
```

Comment:
what happens with the passphrase if it does not get discarded? Is it saved somewhere  ? ...

Reply:
Nope. Actually, this is to discard the passphrase from the LUKS device, as in removing it from a keyslot in the device. In the end, you will end up with the passphrase added by the binding, and this passphrase you provided (which was valid for the device) will not be a valid passphrase anymore.

Reply:
If it does not get discarded, we do not remove it from the LUKS keyslot

Reply:
Even with this explanation, it is hard to understand what happens, and impossible from the description given in the file or in the README. The phrase "specifies whether we should discard the passphrase provided -- via either pass or keyfile, in devices from the LUKS device, after completing the binding operation." is really hard to parse.

Reply:
What about something like this: `discard_passphrase`: indicates whether we should discard the passphrase from the LUKS keyslot in the device. The passphrase in question is the one that was provided either via the ` pass` or `keyfile` arguments.

Reply:
Or something with more context, like:

discard_passphrase: this option is about discarding the LUKS passphrase from the keyslot in the device, after a successful binding. Sometimes dummy passphrases are used when first creating LUKS devices, and then they are removed in post installation steps, replaced with stronger passphrases. If this option is set to yes, the passphrase that was provided via the `pass` or `keyfile` arguments will be removed from the LUKS device after the binding operation completes, i.e., they will not be valid anymore. Default is `no`.

In any case, I agree we need some rewording here.

Reply:
So, one of the typical use cases for `discard_passphrase: yes` is at the first creating LUKS device with `dummy passphrases`? Could there be any other good ones?

Reply:
I think that may be _the_ typical case, in which we have automated (e.g kickstart) installs that set up some sort of "default" password that will be changed/removed later on. 

We suggest something like this here: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption_security-hardening#configuring-automated-enrollment-of-luks-encrypted-volumes-using-kickstart_configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption

Reply:
> If this option is set to yes, the passphrase that was provided via the pass or keyfile arguments will be removed from the LUKS device after the binding operation completes

The README does not describe any `pass` or `keyfile` arguments!